### PR TITLE
ref(github-comments): move option toggle to integrations page

### PR DESCRIPTION
**Pull request by cathteng**
Screenshot 2023-07-19 at 14 40 46

### CODE REVIEW

File: static/app/utils/analytics/integrations/index.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ type SingleIntegrationEventParams = {
   already_installed?: boolean;
   // include the status since people might do weird things testing unpublished integrations
   integration_status?: SentryAppStatus;
-  integration_tab?: 'configurations' | 'overview';
+  integration_tab?: 'configurations' | 'overview' | 'features';
```
Comment (AniketDas-Tekky) [inline static/app/utils/analytics/integrations/index.ts:27]: @Dhrumil-Sentry tab name thoughts?

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 2]
```diff
@@ -132,6 +139,31 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
     return this.metadata.features;
   }
 
+  renderTabs() {
+    // TODO: Convert to styled component
+    const {integrationSlug} = this.props.params;
+    const {organization} = this.props;
+    // TODO(cathy): remove feature check
+    const tabs =
+      integrationSlug === 'github' && organization.features.includes('pr-comment-bot')
```
Comment (nhsiehgit) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:148]: huh.
no way to make this more extensible?

Could we have a mapping of enum to features to include or something? 🤷
Or have the features list from the BE response help dictate this somehow?

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 3]
```diff
@@ -132,6 +139,31 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
...
+
+    return (
+      <ul className="nav nav-tabs border-bottom" style={{paddingTop: '30px'}}>
+        {tabs.map(tabName => (
+          <li
+            key={tabName}
+            className={this.state.tab === tabName ? 'active' : ''}
+            onClick={() => this.onTabChange(tabName)}
+          >
+            <CapitalizedLink>{this.getTabDisplay(tabName)}</CapitalizedLink>
```
Comment (nhsiehgit) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:160]: huh. we're capitalizing via css? 🤷 
lol we could also do this via JS right?

**Review by nhsiehgit — APPROVED** [review]
a few nit comments but not blocking

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 4]
```diff
@@ -132,6 +139,31 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
     return this.metadata.features;
   }
 
+  renderTabs() {
+    // TODO: Convert to styled component
+    const {integrationSlug} = this.props.params;
```
Comment (scttcper) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:144]: i think we normally use `provider.key` for this which should be available in this component

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx (hunk 3 above)
Comment (cathteng) [reply static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:160]: i'm not sure lol i copy pasted from the parent component

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx (hunk 2 above)
Comment (cathteng) [reply static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:148]: oh good point. a map would be good

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx (hunk 3 above)
Comment (cathteng) [reply static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:160]: it's a capitalized link

Comment (cathteng) [reply static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:160]: this capitalizes it visually but keeps the text in the source code as lowercase

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 5]
```diff
@@ -292,6 +324,78 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
...
+              </Fragment>
+            ),
+          },
+        ],
+      },
+    ];
+
+    const jsonFormSettings = {
+      features: new Set(organization.features),
+    };
```
Comment (scttcper) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:368]: just pass the props normally

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 6]
```diff
@@ -292,6 +329,78 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
...
```
Comment (scttcper) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:368]: just pass the props normally

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 7]
```diff
@@ -292,6 +329,78 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
...
+                {t(
+                  'Allow Sentry to comment on pull requests about issues impacting your app.'
+                )}
+              </Fragment>
+            ),
+            disabled: !hasIntegration || !hasOrgWrite,
+            disabledReason: (
+              <Fragment>
+                {t('You must have a GitHub integration to enable this feature.')}
+              </Fragment>
```
Comment (scttcper) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:359]: unused fragment

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 8]
```diff
@@ -292,6 +329,78 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
...
+            name: 'githubPRBot',
+            type: 'boolean',
+            label: t('Enable Pull Request Bot'),
+            visible: ({features}) => features.has('pr-comment-bot'),
+            help: (
+              <Fragment>
+                {t(
+                  'Allow Sentry to comment on pull requests about issues impacting your app.'
+                )}
+              </Fragment>
```
Comment (scttcper) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:353]: same here

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 9]
```diff
@@ -292,6 +329,78 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
...
+    };
+
+    return (
+      <Form
+        data-test-id="organization-settings"
+        apiMethod="PUT"
+        apiEndpoint={endpoint}
+        saveOnBlur
+        allowUndo
+        initialData={organization}
```
Comment (scttcper) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:377]: i'm not sure this makes sense? Shouldn't this be something else

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx [hunk 10]
```diff
@@ -292,6 +329,78 @@ class IntegrationDetailedView extends AbstractIntegrationDetailedView<
...
+      },
+    ];
+
+    const jsonFormSettings = {
+      features: new Set(organization.features),
+    };
+
+    return (
+      <Form
+        data-test-id="organization-settings"
```
Comment (scttcper) [inline static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:372]: remove this if its unused

File: static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx (hunk 9 above)
Comment (cathteng) [reply static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx:377]: if i remove it, it doesn't fetch the org option data correctly to populate the toggle

**Review by scttcper — APPROVED** [review]

**Pull request merged**